freestyle-fs2/shared/src/test/scala/Fs2Tests.scala

```diff
@@ -0,0 +1,101 @@
+package freestyle
+
+import org.scalatest._
+import _root_.fs2._
+import _root_.fs2.util.Free
+import cats.{Applicative, Eval}
+
+import fs2._
+
+import freestyle.fs2._
+import freestyle.implicits._
+import freestyle.fs2.implicits._
+import cats.instances.future._
+import scala.concurrent._
+import scala.concurrent.duration._
+
+class Fs2Tests extends AsyncWordSpec with Matchers {
+  implicit override def executionContext = ExecutionContext.Implicits.global
+
+  "Fetch Freestyle integration" should {
+    import algebras._
+
+    "allow a stream to be interleaved inside a program monadic flow" in {
+      val program = for {
+        a <- app.nonStream.x
+        b <- app.streamM.runLog(Stream.emit(40))
+        c <- Applicative[FreeS[App.T, ?]].pure(1)
+      } yield a + b.head + c
+      program.exec[Future] map { _ shouldBe 42 }
+    }
+
+    "allow a stream to be folded inside a program monadic flow" in {
+      val program = for {
+        a <- app.nonStream.x
+        b <- app.streamM.runFold(0, (x: Int, y: Int) => x + y)(Stream.emits(List(39, 1, 1)))
+      } yield a + b
+      program.exec[Future] map { _ shouldBe 42 }
+    }
+
+    "allow an empty stream to be consumed inside a program monadic flow" in {
+      val program = for {
+        a <- app.nonStream.x
+        b <- app.streamM.runLast(Stream.emits(List.empty[Int]))
+      } yield b
+
+      program.exec[Future] map { _ shouldBe None }
+    }
+
+    "allow a stream to be consumed for its last element inside a program monadic flow" in {
+      val program = for {
+        a <- app.nonStream.x
+        b <- app.streamM.runLast(Stream.emits(List(1, 2, 41)))
+      } yield b.fold(0)(_ + a)
+
+      program.exec[Future] map { _ shouldBe 42 }
+    }
+
+    "allow a stream to be run for its effects inside a program monadic flow" in {
```

Comment:
Once we have the syntax implemented we should also add a test for it to ensure implicits work and the so does the syntax